### DISCUSSION TITLE
How to build ginkgo in an intranet environment

### DISCUSSION THREAD

**Discussion by QuasarJSPark** [ginkgo-project/ginkgo#1506] [Q&A] [answered]
Hello. Since the Linux environment I'm trying to use ginkgo is a secure environment with no internet access, when I run the build script, I can't proceed with the installation anymore because of the git clone option.

How can I use ginkgo in an environment without internet access?

**Comment by upsj** [comment, marked as answer]
The only `git clone` operations necessary are those for third-party dependencies like googletest, nlohmann-json and gflags. If you only want to build Ginkgo, you need none of those, just disable the corresponding components with `-DGINKGO_BUILD_BENCHMARKS=OFF -DGINKGO_BUILD_TESTS=OFF` flags in CMake. If you want to run the tests, you have to build `googletest` yourself from source and install it in a prefix location that you specify via `-DCMAKE_PREFIX_PATH=...` to the Ginkgo build, or use the version from your system package manager, at least version 1.10.0.

**Reply by QuasarJSPark** [reply]
Thank you. Your help made the build successful.
